slick/src/main/scala/slick/jdbc/MacroTreeBuilder.scala

```diff
@@ -16,9 +16,23 @@ private[jdbc] class MacroTreeBuilder[C <: Context](val c: C)(paramsList: List[C#
   lazy val rawQueryParts: List[String] = {
     //Deconstruct macro application to determine the passed string and the actual parameters
     val Apply(Select(Apply(_, List(Apply(_, strArg))), _), paramList) = c.macroApplication
-    strArg map {
-      case Literal(Constant(x: String)) => x
-      case _ => abort("The interpolation contained something other than constants...")
+    val list = strArg map { str =>
+      evalString(str).getOrElse(abort("The interpolation contained something other than constants..."))
```

Comment:
I think the term literal is key here, values can be constant, but are still not allowed.